packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts

```diff
@@ -245,78 +246,61 @@ export function canHaveRestrictedValues(
-    oldColumnModelId2Model.set(columnModel.id, columnModel)
-  }
-
-  // The newSchema will match the proposed schema, but column models that have been changed will have the ID stripped
-  let newSchema: SetOptional<ColumnModel, 'id'>[] = []
-  for (const m of proposedSchema) {
-    const copy: SetOptional<ColumnModel, 'id'> = {
-      ...m,
-    }
-    if (copy.id != null) {
-      // any changes to the existing column model?
-      const oldColumnModel: ColumnModel | undefined =
-        oldColumnModelId2Model.get(copy.id)
-      if (oldColumnModel != undefined && !isEqual(oldColumnModel, copy)) {
-        delete copy.id
+export function transformFormDataToColumnModels(
+  formData: ColumnModelFormData[],
+): SetOptional<ColumnModel, 'id'>[] {
+  return formData.map(
+    (formEntry: ColumnModelFormData): SetOptional<ColumnModel, 'id'> => {
+      // Remove the isSelected field because it was only used for the UI.
+      const { isSelected, ...rest } = formEntry
+      const columnModel = rest as SetOptional<ColumnModel, 'id'>
+      if (columnModel.jsonSubColumns) {
+        columnModel.jsonSubColumns = (
+          columnModel.jsonSubColumns as JsonSubColumnModelFormData[]
+        ).map(
+          (
+            jsonSubColumnFormData: JsonSubColumnModelFormData,
+          ): JsonSubColumnModel => {
+            // isSelected field from the subcolumn for the same reason
+            const { isSelected, ...rest } = jsonSubColumnFormData
+            return {
+              ...rest,
+            }
+          },
+        )
       }
-    }
-    newSchema.push(copy)
-  }
-  const createdColumnModels = (await createColumnModels(accessToken, newSchema))
-    .list
-
-  // now that all columns have been created, figure out the column changes (create, update, and no-op)
-  // keep track of column ids to figure out what columns were deleted.
-  const changes: ColumnChange[] = []
-  const columnIds: Set<string | null> = new Set()
-  for (let i = 0; i < proposedSchema.length; i++) {
-    const oldColumnId: string | null = proposedSchema[i].id ?? null
-    const newColumnId: string | null = createdColumnModels[i].id ?? null
-    columnIds.add(oldColumnId)
-    columnIds.add(newColumnId)
-    if (oldColumnId !== newColumnId) {
-      changes.push({ oldColumnId, newColumnId })
-    }
-  }
+      return columnModel
+    },
+  )
+}
 
-  // delete columns that are not represented in the changes already (create or update)
-  for (const oldColumnModel of oldSchema) {
-    const oldColumnId: string | null = oldColumnModel.id
-    if (!columnIds.has(oldColumnId)) {
-      changes.push({ oldColumnId, newColumnId: null })
+/**
+ * Transform ColumnModels returned by Synapse into the form data for the TableColumnSchemaForm.
+ * Default column models, if available, are used to determine which columns should not be editable in the form.
+ * @param columnModels
+ * @param defaultColumns
+ */
+export function transformColumnModelsToFormData(
```

Comment:
This is new, and is used to mark if a column is a 'default column' so we know to not let the user modify the name.